telingo/theory/path.py

```diff
@@ -0,0 +1,71 @@
+class Path(object):
+    """
+    Base class of all path
+
+    Members:
+    __rep  -- unique string representation of the path
+    """
+    def __init__(self, rep):
+        """
+        Initializes a formula with the given string representation.
+        """
+        self.__rep  = rep
+
+    @property
+    def _rep(self):
+        """
+        Return the unique string representaiton of the formula.
+        """
+        return self.__rep
+
+class SkipPath(Path):
+    def __init__(self):
+        Path.__init__(self, "(&skip)")
+
+class BinaryPath(Path):
+    """
+    Members:
+    _lhs
+    _rhs
+
+    """
+    def __init__(self, rep, lhs, rhs):
+        self._lhs = lhs
+        self._rhs = rhs
+        Path.__init__(self, rep)
+
+class ChoicePath(BinaryPath):
+    
+    def __init__(self, lhs, rhs):
+        BinaryPath.__init__(self, "({}{}{})".format(lhs._rep,"+",rhs._rep), lhs, rhs)
```

Comment:
Please put a space after `,` and also around binary operators, e.g., `a + b` instead of `a+b`. Right now we have a mix of different styles. Please do this for all files. I would also suggest to not pass string literals into format. Writing
```python
"({}+{})".format(lhs._rep, rhs._rep)
```
is better.